server/words.js

```diff
@@ -8,53 +8,57 @@ const easy = ['cat', 'dog', 'umbrella', 'mouse', 'apple', 'banana', 'pineapple']
 const medium = ['merry-go-round', 'polaroid', 'photo album']
 const hard = ['high school', 'carnival', 'castle']
 const impossible = ['mystery', 'orange', 'novel', 'regret', 'excited']
-const custom = {}
+// const custom = {}
 
-const words = { easy, medium, hard, impossible, custom }
+const DEFAULT_WORDS = { easy, medium, hard, impossible, custom: [] }
+
+const wordSets = {}
 
 function addCustomWord(roomCode, word) {
-  if (Object.prototype.hasOwnProperty.call(custom, roomCode)) {
-    if (!custom[roomCode].find((current) => current === word)) {
-      custom[roomCode] = [...custom[roomCode], word]
-    }
-    console.log('if room already exists', Object.values(custom))
-  } else {
-    custom[roomCode] = [word]
-    console.log('room doesnt exists', Object.values(custom))
+  if (!wordSets[roomCode].custom.includes(word)) {
+    wordSets[roomCode].custom = [...wordSets[roomCode].custom, word]
   }
+  console.log('if room already exists', Object.values(wordSets[roomCode].custom))
+}
+
+function addWordSet(roomCode) {
+  wordSets[roomCode] = JSON.parse(JSON.stringify(DEFAULT_WORDS))
 }
 
-function clearCustomWords(roomCode) {
-  delete custom[roomCode]
-  console.log(custom[roomCode])
+function clearWordSet(roomCode) {
+  delete wordSets[roomCode]
+  console.log(wordSets[roomCode])
 }
 
 function getRandom(list) {
   return list[Math.floor(Math.random() * list.length)]
 }
 
-function getRandomDifficulty() {
-  return getRandom(Object.keys(words))
+function getRandomDifficulty(roomCode) {
+  return getRandom(Object.keys(wordSets[roomCode]))
 }
 
 function getRandomWord(difficulty, roomCode) {
-  let wordList
-  if (difficulty === 'custom') {
-    if (!Object.prototype.hasOwnProperty.call(custom, roomCode)) {
-      wordList = words[getRandomDifficulty()]
-    } else {
-      wordList = custom[roomCode]
-    }
-  } else {
-    wordList = words[difficulty]
+  let newDifficulty = difficulty
+  if (!wordSets[roomCode][difficulty]) {
+    newDifficulty = getRandomDifficulty(roomCode)
+  }
+  const wordList = wordSets[roomCode][newDifficulty]
+  const word = getRandom(wordList)
+  const idx = wordSets[roomCode][newDifficulty].indexOf(word)
+  wordSets[roomCode][newDifficulty].splice(idx, 1)
+  console.log(wordSets)
+  if (wordSets[roomCode][newDifficulty].length < 1) {
```

Comment:
Could we also have this check at the top just incase they didnt put in any custom words, then we can delete the custom difficulty

Reply:
Maybe we can call the variable 'index' to be most explicit.

Reply:
I don't think we can because the wordSet is created before any custom words are added so it will be initialized to: 
`wordSets[roomCode] = { easy, medium, hard, impossible, custom: [] }`
it also shouldn't matter if custom is [], because if custom is empty and it's selected as the difficulty then it will just delete custom at that time.

Reply:
wait but in the case of: 
```
wordSets[roomCode] = { easy, medium, hard, impossible, custom: [] }
```
 the condition here if difficulty is custom
```
if (!wordSets[roomCode][difficulty]) 
```
Should be false since the [] in javascript is truthy, so it wont enter the if block. Then it will try to getRandom from an empty list, so I think we need to delete the custom before doing this check
``` if (!wordSets[roomCode][difficulty]) ```

Reply:
unless im misunderstanding something? is there another place that we check if custom is empty and delete it besides in this method?

Reply:
I misundersstood, it should be fixed now. There is no custom difficulty in the default object and it will only be added to wordSets[roomCode] when the user adds the first custom word.

Reply:
wait this might be too late but if we're just going to have custom words be specific to one instance of a game maybe we should have made that a property of the game object instead of having it be in words.js LOL but maybe not idk